src/sys/windows.rs

```diff
@@ -920,6 +920,21 @@ impl crate::Socket {
             Ok(())
         }
     }
+
+    /// Returns the [`Protocol`] of this socket by checking the `SO_PROTOCOL_INFOW`
+    /// option on this socket.
+    #[cfg(feature = "all")]
+    pub fn protocol(&self) -> io::Result<Option<crate::Protocol>> {
+        use windows_sys::Win32::Networking::WinSock::SO_PROTOCOL_INFOW;
```

Comment:
Nit: can you move this to the other imports.

Reply:
Moved.